### ISSUE TITLE
Search with semicolon gives no result. Search differs from earlier versions of find()(1.7.6.x vs 1.7.7.x)

### ISSUE THREAD

**Issue by venditdevs** [PrestaShop/PrestaShop#25394]
In Prestashop 1.7.6.X you could search for multiple products based on (e.g.) tags. You could split them with a semicolon.
So if i searched for  testproduct1; testproduct5; testproduct8 it would find all 3 products.

Now, we've found out on Prestashop 1.7.7.5 ( this is the version where we noticed the problems so it could be earlier , we upgraded from 1.7.6.8 to 1.7.7.5 ). 
If we search for testproduct1 we get 1 result as expected. 
If we search for testproduct1; testproduct2 the result is 'NO PRODUCTS FOUND'  but it SHOULD  give us the 2 products
If we search for testproduct1; testproduct2; testproduct3 the result is 'NO PRODUCTS FOUND'  but it SHOULD  give us the 3 products

If i replace the find() function on Prestashop 1.7.7.5 with the one from 1.7.6.8 it all works as expected and fine

* PrestaShop version: 1.7.7.5 
* PHP version: PHP 7.3

NOTE : we also found a shop with 1.7.7.2 . Same problem

**Comment by florine2623** [comment]
Hello @venditdevs ,

I was able to reproduce the issue with PrestaShop version 1.7.7.5 and 1.7.8.x, so I can confirm that this behavior is indeed a bug.

Here are the screenshots :
In PS 1.7.7.5 -> NOK:
![Screenshot 2021-07-21 at 15 51 55](https://user-images.githubusercontent.com/16019289/126500257-20df953d-cb2c-4858-a16e-abde53dad49b.png)

In PS 1.7.6.9 -> OK:
![Screenshot 2021-07-21 at 15 51 13](https://user-images.githubusercontent.com/16019289/126500162-593faa76-2353-43b5-821f-389f783844a6.png)

Please be aware that due to the high number of reported issues, some bugs might take a very long time to be resolved. If this issue is important to you and you cannot wait for it to be fixed on the project’s own time, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you. If you manage to get the problem fixed, then please contribute the fix back to the project as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/). Remember that the more people contribute, the better PrestaShop becomes for everyone.

Thank you

**Comment by venditdevs** [comment]
Hi, 

I allready found the problem.
It's this line: $eligibleProducts2 = array_intersect($eligibleProducts2, $productIds);
This will rerurn the differences between the two arrays which is none when you use 2 or more search words and so the result is empty. 

Looking into it for a solution

**Comment by venditdevs** [comment]
@florine2623 : It's fixed. I created PR.

**Comment by hibatallahAouadni** [comment]
Hello @venditdevs 

Thanks for your PR 🚀

**Comment by Multipage** [comment]
@khouloudbelguith @florine2623 
In the release notes it says:
`Search with semicolon gives no result. Search differs from earlier versions of find()`

Updated a shop..fix not present...downloaded the release in zip ...fix not present in zip file!

**Comment by Multipage** [comment]
@khouloudbelguith @florine2623 please respond!

**Comment by khouloudbelguith** [comment]
Hi @Multipage,

What is the exact version PrestaShop did you use?

Thanks!

**Comment by Multipage** [comment]
@khouloudbelguith We've updated with 1-click upgrade to 1.7.8.1 and found out the fix wasn't there. So we've downloaded the file through the Prestashpop website again ( again version 1.7.8.1) and looked in the zip for the fix. Fix isn't in the zipfile either.
Something is going wrong with building the update files

**Comment by khouloudbelguith** [comment]
@Multipage, the issue is fixed in the next version PS8.0.0
You can check the changed files here: https://github.com/PrestaShop/PrestaShop/pull/25396/files.

Thanks!

**Comment by Multipage** [comment]
> @Multipage, the issue is fixed in the next version PS8.0.0 You can check the changed files here: https://github.com/PrestaShop/PrestaShop/pull/25396/files.
> 
> Thanks!

@khouloudbelguith Why is it listed in the release notes op 1.7.8.1 then?

https://build.prestashop.com/news/prestashop-1-7-8-1-maintenance-release/
Under Front office regressions
```
Search with semicolon gives no result. Search differs from earlier versions of find()
```

**Comment by matks** [comment]
@Multipage @khouloudbelguith After looking carefully at this topic, it seems a mistake was made.

This bug was fixed in PrestaShop 8.0.0 and should not have been added to the 1.7.8.1 Release note.

I will fix the Release Note to remove this line.
